### PR TITLE
Configure VolumeMounts with subPath in ContainerBuilder

### PR DESCRIPTION
**Pull request by tommaso-borgato**
This MR introduces the option to configure volumeMounts with subPath through cz.xtf.builder.builders.pod.ContainerBuilder

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Code is self-descriptive and/or documented

### CODE REVIEW

**Review by fabiobrz — APPROVED** [review]
LGTM - @mnovak1 this seems conservative to me, since the only relevant change keeps the compatibility with the existing implementation. WDYT?

**Review by jbliznak — APPROVED** [review]

File: builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java [hunk 1]
```diff
@@ -271,30 +277,6 @@ public void addPreStopHandler(Handler handler) {
         preStopHandler = handler;
     }
 
-    private static class VolumeMount {
```
Comment (simkam) [inline builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: I would keep changes minimal, it is used only in this class, I would keep it nested and not add any unnecessary changes. 

Definitely, making the new class public makes it part of public API which we don't want. Unless I'm missing something.

**Review by simkam — CHANGES_REQUESTED** [review]

File: builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java (hunk 1 above)
Comment (tommaso-borgato) [reply builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: unfortunately we need this class to be public in order to configure its `subPath` property, e.g.:
```
volumeMounts.add(
				new VolumeMount("data-dir", "/opt/server/standalone/data/ejb-xa-recovery", false, "ejb-xa-recovery"));
```

Comment (simkam) [reply builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: Can you please elaborate? I still don't see it. Isn't it
```
ContainerBuilder.addVolumeMount("data-dir", "/opt/server/standalone/data/ejb-xa-recovery", false, "ejb-xa-recovery")
```
?

Comment (tommaso-borgato) [reply builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: it's actually in another project

Comment (tommaso-borgato) [reply builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: I've sent you the link on the chat since it's an internal repo

Comment (tommaso-borgato) [reply builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: btw, in the snipped you pasted there a missing `new VolumeMount(...`

Comment (simkam) [reply builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: ok, looking at other code, now I understand ... but in this case, I would even extend it by one more signature 

```
public ContainerBuilder addVolumeMount(VolumeMount volumeMount) {
   this.volumeMounts.add(volumeMount);
   return this;
}
```

otherwise it doesn't make sense to me to make it part of public API because there is really no way to use it outside XTF.

Comment (tommaso-borgato) [reply builder/src/main/java/cz/xtf/builder/builders/pod/ContainerBuilder.java:274]: done! tnks for the suggestion

**Review by simkam — APPROVED** [review]
